worker.js

```diff
@@ -0,0 +1,30 @@
+self.onmessage = function(e) {
+  sendPacket(e.data.url, e.data.userAgent, self.postMessage);
+};
+
+function sendPacket(url, userAgent, callback)
+{
+    var request = new XMLHttpRequest();
+    request.open('GET', url);
+    request.setRequestHeader('Access-Control-Allow-Origin', '*');
+    request.setRequestHeader('Access-Control-Allow-Methods', 'GET, POST, PATCH, PUT, DELETE, OPTIONS');
+    request.setRequestHeader('Access-Control-Allow-Headers', 'Origin, Content-Type, X-Auth-Token');
+    
+    // TODO: investigate about setting user agent
```

Comment:
This seems to be impossible now:
https://stackoverflow.com/questions/1307013/mocking-a-useragent-in-javascript